interfaces/builtin/bcache.go

```diff
@@ -0,0 +1,30 @@
+package builtin
+
+const bcacheSummary = `allows access to bcache kernel interfaces`
+
+const bcacheBaseDeclarationSlots = `
+  bcache:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+const bcacheConnectedPlugAppArmor = `
+# Description: Can access kernel bcache interfaces in sysfs
+
+# Allow read/write access to kernel bcache interfaces in sysfs
+# includes items in /sys/fs/bcache, /sys/block and /sys/devices
```

Comment:
`/sys**bcache**` seems far too non-specific - can we instead change this to include more limited / known paths?

Also from what I can find in https://bcachefs.org/bcachefs-principles-of-operation.pdf there are often paths like:

`/sys/fs/bcachefs/<UUID>/options/<option>`  

and 

`/sys/fs/bcachefs/<UUID>/time_stats/<stat>`

so I would instead prefer to see something like the following (note I am not sure what the format of UUID is so the `[0-9]*` part will likely be incorrect):

`/sys/fs/bcachefs/[0-9]*/options/{block_size,btree_node_size,...} rwk`
`/sys/fs/bcachefs/[0-9]*/time_stats/{blocked_allocate,...} rwk`


Also since you mention paths under `/sys/block` would this be covered by the existing block-devices interface? Or put another way - does block-devices need updating to include some relevant rules to cover the devices presented via bcachefs?

Reply:
Also does the interface require any capabilities like `CAP_SYS_ADMIN` when interfacing with it?

Reply:
To be clear, my requirement for submitting this PR is not for bcachefs, it is only related to bcache ([pure block device cache](https://www.kernel.org/doc/Documentation/bcache.txt)). It operates only on block devices, but once a bcache block device is created there are several interfaces in /sys to configure and modify caches. So my app works with block devices interface but also needs these extra bits to allow access to those interfaces in /sys.

To reiterate, I don't need /sys/fs/bcachefs/*.  Some other app may need bcachefs in future and it could extend this interface or a new specific (bcachefs) interface could be created at that time.

I did some further testing and I think this might be the minimum requirements:
/sys/fs/bcache/ r,
/sys/fs/bcache/[0-9a-z]*-[0-9a-z]*-[0-9a-z]*-[0-9a-z]*/* rw,
/sys/fs/bcache/register w,
/sys/devices/**/bcache/** rw,

No need for CAP_SYS_ADMIN that I have encountered.

Is this specific enough?

Or alternatively, do you think these apparmor settings should be added to the block devices interface, since bcache operations are kind of a subset of block devices?

Reply:
Yes I agree, thanks for the clarification @rafalop - apologies I misinterpreted it from the beginning - this should likely just be part of the existing block-devices interface.